dimensiondata/src/main/java/org/jclouds/examples/dimensiondata/cloudcontrol/DeleteServerVlanAndNetworkDomain.java

```diff
@@ -0,0 +1,203 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.examples.dimensiondata.cloudcontrol;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import org.jclouds.ContextBuilder;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Server;
+import org.jclouds.dimensiondata.cloudcontrol.domain.TagKey;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Vlan;
+import org.jclouds.dimensiondata.cloudcontrol.options.DatacenterIdListFilters;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+
+import static org.jclouds.examples.dimensiondata.cloudcontrol.WaitForUtils.*;
+
+/**
+ * This class will attempt to delete the assets created in org.jclouds.examples.dimensiondata.cloudcontrol.DeployNetworkDomainVlanAndServer:
+ * <ol>Server</ol>
+ * <ol>Vlan</ol>
+ * <ol>Network Domain</ol>
+ * <ol>Tag Key</ol>
+ */
+public class DeleteServerVlanAndNetworkDomain
+{
+    private static final String AU_9 = "AU9";
+    private static final String DIMENSIONDATA_CLOUDCONTROL_PROVIDER = "dimensiondata-cloudcontrol";
+
+    public static void main(String[] args)
+    {
+        /*
+         * Build an instance of the Dimension DataCloud Control Provider using the endpoint provided.
+         * Typically the endpoint will be of the form https://api-GEO.dimensiondata.com/caas
+         * We also need to provide authenticate details, a username and password.
+         *
+         * Internally the Dimension Data CloudControl Provider will use the org.jclouds.dimensiondata.cloudcontrol.features.AccountApi
+         * to lookup the organization identifier so that it is used as part of the requests.
+         *
+         */
+        String endpoint = args[0];
+        String username = args[1];
+        String password = args[2];
+        ContextBuilder contextBuilder = ContextBuilder.newBuilder(DIMENSIONDATA_CLOUDCONTROL_PROVIDER);
+        DimensionDataCloudControlApi api = contextBuilder
+                .endpoint(endpoint)
+                .credentials(username, password)
+                .modules(ImmutableSet.<Module>of(new SLF4JLoggingModule()))
+                .buildApi(DimensionDataCloudControlApi.class);
+
+        /*
+         * Retrieve the Guice injector from the context.
+         * We will use this for retrieving the some Predicates that are used by the following operations.
+         */
+        Injector injector = contextBuilder.buildInjector();
```

Comment:
Don't do this, as you're building the context twice. Instead:

```java
ApiContext<DimensionDataCloudControlApi> ctx = contextBuilder
   .endpoint(endpoint)
   .credentials(username, password)
   .modules(ImmutableSet.of(new SLF4JLoggingModule()))
   .build();

Injector injector = ctx.utils().injector();
DimensionDataCloudControlApi api = ctx.getApi();
```

Also remember to always close the `ctx` in a `finally` clause.